src/middlewared/middlewared/plugins/reporting/events.py

```diff
@@ -2,12 +2,63 @@
 
 from middlewared.event import EventSource
 from middlewared.schema import Dict, Float, Int
+from middlewared.service import Service
 from middlewared.utils.disks import get_disk_names, get_disks_with_identifiers
 from middlewared.validators import Range
 
 from .realtime_reporting import get_arc_stats, get_cpu_stats, get_disk_stats, get_interface_stats, get_memory_info
 
 
+class ReportingRealtimeService(Service):
+
+    class Config:
+        namespace = 'reporting.realtime'
+        private = True
+
+    def stats(self, disk_mapping=None):
```

Comment:
Sorry but no. This is a copy and paste of the realtime event source. We need to figure out how to avoid this.

Reply:
Oh, I see what you did! Ignore me.